### ISSUE TITLE
Selecting a shared folder or external storage as download location does not work

### ISSUE THREAD

**Issue by Larx** [e-alfred/ocdownloader#30]
As stated in the subject:
I try to either
- set a folder which has been shared to me, or
- an external folder (NAS, mounted via FTP) 
as location for the Downloads.

Both variants do not work.

**Comment by e-alfred** [comment]
Do you see something in the logs actually? Which downloader did you use? Aria2 is an external tool that can't access these directories because they are not mounted directly on the filesystem level of the operating system actually.

**Comment by Larx** [comment]
Sorry, I did not think of getting a log, will try the next time.
If I use the default setup (with Aria2), a folder "Download" will be created in my Nextcloud instance (this is the preset in the settings). And then Aria2 uses this folder, so it seems to be able to access the directory.
Maybe I understand something wrong: The Download folder as given in ocDownloader settings is that the Nextcloud folder (it seems to be when using "/Downloads"), or is it a folder on the local filesystem as seen by Aria2?

**Comment by GAS85** [comment]
Following share scenario does not work:
1. User A start download
 - folder "Downloads" will be created and file A will be located there.
   Folder is created under nextcloud/data/UserA/files/Downloads
2. User A share "Downloads" with User B
3. User B can see file A in "Downloads"
4. User B start downloading file B.zip by his own and expected that file will be located in the same shared Folder.
 - User A nether Uaser B will never see this file B.zip --> File will be wrongly located in nextcloud/data/UserA/B.zip folder and will never be indexed.

Would be nice to put file in Downloads where it is really located.

**Comment by TokoAd** [comment]
I also encountered this problem. I used external storage to expand my data space. The ocDownloader seems to be unable to download to storage other than "local". I used amazon S3 and FTP for testing. The added download task will not run. . shows "waiting for download".

I found it under the log. I don't know if it's related to the problem I'm having.

`call_user_func_array() expects parameter 1 to be a valid callback, class 'OCA\Files_External\Lib\Storage\AmazonS3' does not have a method 'getLocalFolder' at /var/www/html/icit/lib/private/Files/Storage/Wrapper/Wrapper.php#505`

@e-alfred Can ocDownloader implement "Download to External Storage" in the future?

**Comment by pdschandler** [comment]
Yeah, my company uses external storage in Nextcloud almost exclusively, as a way to access our centralized storage remotely. Setting the download location to one of these shared folders would allow us to download assets delivered to us by clients in the background. Then other employees and application would see them available.
